### ISSUE TITLE
parameter averaging

### ISSUE THREAD

**Issue by romikforest** [explosion/spaCy#2745]
You recommend (https://spacy.io/usage/training#section-tips):

> Then when we want to actually use the model, we use the averages, not the most recent value. 
> with nlp.use_params(optimizer.averages):
>     nlp.to_disk('/model')

But in the classification example https://spacy.io/usage/training#section-textcat model is saved only not in nlp.use_params(optimizer.averages) scope. Is it correct?

**Comment by honnibal** [comment]
You're right, that's a bug in the example. Normally the best thing to do is to save the model with the best accuracy each epoch, when we do the evaluation, and then use the best model. Using the model from the last epoch is usually worse --- especially without the parameter averaging.
